app/app.py

```diff
@@ -0,0 +1,301 @@
+
+@app.route('/get_pdf_id_imgs', methods=['GET'])
+def get_pdf_id_imgs():
+    pdf_id = request.args.get('pdf_id')
+    sql = '''select pdf_id, img_id, pdf_page from imgs where pdf_id = '''+br(pdf_id)+''' order by img_id '''
+    data = select(sql)
+    html = ''
+    for pdf_id, img_id, pdf_page in data: 
+        html += f'<img id="{img_id}" class="img-fluid" tyle="width:100%;;" src="/files/imgs/{pdf_id}_{pdf_page}.png"></img>'
+    return html 
+
+# Helper Functions for Development
+# ----------------------------------------------------------------------------------------------
+@app.route('/reset', methods=['GET'])
+def reset(): 
+    try:
+        for folder in ['/app/files/pdfs/', '/app/files/imgs/']: 
+            print('folder: ', folder)
+            for file in [f for f in listdir(folder) if isfile(join(folder, f))]:  os.remove(os.path.join(folder, file))
+        
+        for table in ['pdfs', 'imgs', 'api_keys']: 
+            print('table: ', table)
+            delete('''drop table '''+table+'''; ''')
+    
+        sql_tables = open ('/app/db/tables.sql', "r").read().split(';')
+        for sql in sql_tables: 
+            print(sql)
+            insert(sql)
+
+        return app.response_class(response=json.dumps({"OK": "200 - Everything was reset"}),status=200,mimetype='application/json')
+    except Exception as e:
+        return app.response_class(response=json.dumps({"error":"500  - Internal Server Error "}),status=500,mimetype='application/json')
+
+@app.route('/folder/', methods=['GET'])
+@app.route('/folder/<folder>', methods=['GET'])
+def folder(folder=''):
+    path = '/app/files/'+str(folder)
+    files_in_folder = [f for f in listdir(path) if isfile(join(path, f))]
+    return str(files_in_folder)
+
+@app.route('/job_info/<job_id>', methods=['GET'])
+def job_info(job_id):
+    try:
+        job = rq.worker.Job.fetch(job_id, connection=worker.rq.connection)
+        return str(job)
+    except Exception as e:
+        return str(e)
+
+@app.route('/delete_job/<job_id>', methods=['GET'])
+def delete_job(job_id):
+    try:
+        job = rq.job.Job.fetch(job_id, connection=jobs.rq.connection)
+        job.delete()  
+    except Exception as e:
+        return str(e)
+
+# API 
+# ----------------------------------------------------------------------------------------------
+@app.route('/documents',methods = ['POST'])
+@app.route('/documents/<document_id>',methods = ['GET'])
+@app.route('/documents/<document_id>/pages/<number>',methods = ['GET'])
+def documents(document_id='', number=''):
+    api_key = ''
+    if 'api-key' in request.headers: 
+        api_key = request.headers['api-key']
+    else:
+        if 'api-key' in request.args:
+            api_key = request.args['api-key']
+        else:
+            json_data = {"error":"401 - Unauthorized - API Key not received in Header or as URL parameter"}
+            return app.response_class(response=json.dumps(json_data),status=401,mimetype='application/json')
```

Comment:
the authentication procedure is better off when done in a separate function. In Flask, the authentication can be utilized either as a decorator, or using a decorator

Reply:
I agree a decorator would be a more elegant solution.

Reply:
As far as I'm concerned, it's a de facto a standard approach. For instance, Miguel Grinberg, the author of Mega Flask tutorial, uses it https://blog.miguelgrinberg.com/post/restful-authentication-with-flask.